addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -113,13 +119,31 @@ def _compute_ongoing_sessions_count(self):
         "user_ids.livechat_ongoing_session_count",
     )
     def _compute_remaining_session_capacity(self):
+        agent_partner_ids = self.mapped('user_ids.partner_id').ids
+        count_by_agent_channel = {
+            (operator.id, channel.id): count
+            for operator, channel, count in self.env["discuss.channel"]._read_group(
+                [
+                    ("livechat_end_dt", "=", False),
+                    ("livechat_channel_id", "=", self.ids),
+                    ("livechat_operator_id", "in", agent_partner_ids),
+                ],
+                ["livechat_operator_id", "livechat_channel_id"],
+                ["__count"],
+            )
+        }
         for channel in self:
             total_capacity = channel.max_sessions * len(channel.user_ids)
-            capacity = total_capacity - sum(channel.user_ids.mapped("livechat_ongoing_session_count"))
+            capacity = total_capacity - sum(
```

Comment:
I think we're overcomplicating it. All we really want is to treat agents on a call as having no capacity. We're doing complex things with dictionaries indexed by tuples, sums, subtractions... But couldn't it be much simpler?

- Count the number of ongoing chats per channel, excluding agents who are on a call: this is the capacity to remove.
- Multiply the number of agents in a call by `max_sessions` (if `block_assignment_during_call` is enabled): this is the capacity lost due to agents being unavailable.

And that’s it, right?